util/src/main/java/care/smith/fts/util/fhir/FhirTag.java

```diff
@@ -0,0 +1,45 @@
+package care.smith.fts.util.fhir;
+
+import java.util.Objects;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Bundle.BundleEntryComponent;
+import org.hl7.fhir.r4.model.Coding;
+
+public interface FhirTag {
+
+  /**
+   * Adds a specified tag to every resource in a HAPI FHIR Bundle.
+   *
+   * @param bundle The HAPI FHIR Bundle to process
+   * @param system The tag coding system
+   * @param code The tag code
+   * @param display The tag display text (optional)
+   * @return The modified Bundle with tags added to every resource
+   */
+  static Bundle addTagToAllResources(Bundle bundle, String system, String code, String display) {
+    if (bundle == null) {
+      return null;
+    }
+
+    var tagCoding = new Coding().setSystem(system).setCode(code);
+    if (display != null && !display.isEmpty()) {
+      tagCoding.setDisplay(display);
+    }
+
+    var bundleMeta = bundle.getMeta();
+    bundleMeta.addTag(tagCoding);
+    bundle.setMeta(bundleMeta);
+
+    bundle.getEntry().stream()
+        .map(BundleEntryComponent::getResource)
+        .filter(Objects::nonNull)
```

Comment:
Do we need to filter for nonNull? Is it possible to set a null resource into a BundleEntryComponent?

Reply:
Unfortunately, yes.